overlord/patch/patch6_2.go

```diff
@@ -112,23 +112,40 @@ func patch6_2(st *state.State) error {
 		return fmt.Errorf("internal error: cannot get snaps: %s", err)
 	}
 
-	// Migrate snapstate
-	for name, raw := range snaps {
+	var hasSnapdSnap bool
```

Comment:
I'm probably going to reveal my golang ignorance now, but don't we need to initialize this to false?

Reply:
No, go initializes bool to false (and other types to their default values).